### PR TITLE
DOC: Factory image and flashing cards from scratch

### PR DESCRIPTION
**Pull request by jsvogt**
How to build the factory image, the mcs file, and finally flash an card from scratch using the Xilinx JTAG adapter

### CODE REVIEW

**Comment by jsvogt** [comment]
Added the FPGA device type sanity check and corrected the grammar in hardware/Readme.md
=> Now ready to be merged

File: hardware/doc/Bitstream_flashing.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+Programming User Partition with my_user_image.bin
+  Program ->  for Size: 56 in blocks (32K Words or 128K Bytes)
+
+Erasing Flash
+...
+```
+# How to build the factory ("golden") bitstream image
+
+All supported FPGA cards have at least two flash partitions for the FPGA bitstream. The *user* partition is loaded by default on power-on. The FPGA loader will automatically load the *factory* partition if loading the user partition fails. That may happen if the flashing process for the user partition was interrupted, or if the flash device on the card or the bitstream file used for flashing got corrupted.
```
Comment (boekholt) [inline hardware/doc/Bitstream_flashing.md]: Did you repeat this intentionally (see above)?

File: hardware/README.md [hunk 2]
```diff
@@ -87,6 +87,9 @@ This will also build the software tools and the PSLSE which are required to run
 
 Please refer to `$SNAP_ROOT/hardware/Makefile` for more supported targets like clean, gitclean, create_environment, ...
 
+## FPGA bitstream image update
+Please see [hardware/doc/Bitstream_flashing.md](./doc/Bitstream_flashing.md) for instructions on how to update the FPGA bitstream image, build factory images and program cards from scratch.
```
Comment (boekholt) [inline hardware/README.md]: [snap/hardware/doc/Bitstream_flashing.md](...) instead?

File: hardware/doc/Bitstream_flashing.md (hunk 1 above)
Comment (boekholt) [inline hardware/doc/Bitstream_flashing.md]: Did you intentionally repeat this (see above "FPGA bitstream images")?

**Pull request merged**